File: app/src/main/java/com/kode/utair/domain/interactors/WeatherForecastInteractor.java
package com.kode.utair.domain.interactors;

import com.kode.utair.domain.models.WeatherForecastEntity;
import com.kode.utair.domain.models.WeatherSettings;
import com.kode.utair.domain.usecases.GetWeatherForecastUseCase;
import com.kode.utair.domain.usecases.GetWeatherSettingsUseCase;

import javax.inject.Inject;

import io.reactivex.Single;

public class WeatherForecastInteractor {

    private GetWeatherSettingsUseCase getWeatherForecastSettings;
    private GetWeatherForecastUseCase getWeatherForecastUseCase;

    @Inject
    public WeatherForecastInteractor(GetWeatherSettingsUseCase getWeatherForecastSettings, GetWeatherForecastUseCase getWeatherForecastUseCase) {
        this.getWeatherForecastSettings = getWeatherForecastSettings;
        this.getWeatherForecastUseCase = getWeatherForecastUseCase;
    }

    public Single<WeatherSettings> getWeatherSettings() {
        return getWeatherForecastSettings.getSingle();
    }

    public Single<WeatherForecastEntity> getWeatherForecastForCity(String cityName) {
        return getWeatherForecastUseCase.getSingle(cityName);
    }
}

File: app/src/main/java/com/kode/utair/domain/interactors/WeatherForecastInteractor.kt
package com.kode.utair.domain.interactors

import android.util.Log
import com.kode.utair.domain.models.WeatherForecastEntity
import com.kode.utair.domain.models.WeatherSettings
import com.kode.utair.domain.repository.IWeatherRepository
import com.kode.utair.domain.repository.IWeatherSettingsRepository
import io.reactivex.Single
import javax.inject.Inject

class WeatherForecastInteractor @Inject constructor(
        private val weatherSettingsRepository: IWeatherSettingsRepository,
        private val weatherRepository: IWeatherRepository
) {

    fun getWeatherSettings(): Single<WeatherSettings> {
        return weatherSettingsRepository.getWeatherSettings()
    }

    fun getWeatherForecastForCity(cityName: String): Single<WeatherForecastEntity> {
        return weatherRepository.getWeatherForecastForCity(cityName)
    }

}